hbase-server/src/main/java/org/apache/hadoop/hbase/coordination/ZKSplitTransactionCoordination.java

```diff
@@ -222,9 +229,21 @@ public void completeSplitTransaction(final RegionServerServices services, Region
     // Tell master about split by updating zk. If we fail, abort.
     if (coordinationManager.getServer() != null) {
       try {
-        zstd.setZnodeVersion(transitionSplittingNode(parent.getRegionInfo(), a.getRegionInfo(),
-          b.getRegionInfo(), coordinationManager.getServer().getServerName(), zstd,
-          RS_ZK_REGION_SPLITTING, RS_ZK_REGION_SPLIT));
+        int newNodeVersion = -1;
+        for (int i = 0; i < SPLIT_WAIT_TIMEOUT; i++) {
+          newNodeVersion = transitionSplittingNode(parent.getRegionInfo(), a.getRegionInfo(),
+            b.getRegionInfo(), coordinationManager.getServer().getServerName(), zstd,
+            RS_ZK_REGION_SPLITTING, RS_ZK_REGION_SPLIT);
+          if (newNodeVersion != -1) {
+            break;
+          }
+          Thread.sleep(100);
+        }
+        if (newNodeVersion == -1) {
```

Comment:
Throwing an exception here before blindly calling setZnodeVersion is definitely an improvement.